frontend/src/lib/components/PropertyFilters/PropertyFilter.js

```diff
@@ -17,9 +17,18 @@ export function PropertyFilter({ index, onComplete, logic }) {
                     autoFocus={!key}
                     defaultOpen={!key}
                     placeholder="Property key"
-                    value={key}
+                    labelInValue
+                    value={{ key, label: keyMapping[type === 'element' ? 'element' : 'event'][key]?.label || key }}
                     filterOption={(input, option) => option.value?.toLowerCase().indexOf(input.toLowerCase()) >= 0}
-                    onChange={(_, new_key) => setFilter(index, new_key.value, undefined, operator, new_key.type)}
+                    onChange={(_, new_key) =>
+                        setFilter(
+                            index,
+                            new_key.value.replace(/event_|person_|element_/gi, ''),
```

Comment:
Best to hook this to the beginning of the string with a `^` like: `/^(event_|person_|element_)/gi`